kion/resource_azure_policy_test.go

```diff
@@ -0,0 +1,197 @@
+package kion
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	hc "github.com/kionsoftware/terraform-provider-kion/kion/internal/kionclient"
+)
+
+const (
+	resourceTypeAzurePolicy = "kion_azure_policy"
+	resourceNameAzurePolicy = "azure_policy1"
+)
+
+var (
+	initialTestAzurePolicy = hc.AzurePolicyCreate{
+		AzurePolicy: struct {
+			Description string `json:"description"`
+			Name        string `json:"name"`
+			Parameters  string `json:"parameters"`
+			Policy      string `json:"policy"`
+		}{
+			Description: "Initial description",
+			Name:        "Initial Policy",
+			Parameters:  "{}",
+			Policy:      "{}",
+		},
+		OwnerUserGroups: &[]int{1},
+		OwnerUsers:      &[]int{1},
+	}
+	updatedTestAzurePolicy = hc.AzurePolicyCreate{
+		AzurePolicy: struct {
+			Description string `json:"description"`
+			Name        string `json:"name"`
+			Parameters  string `json:"parameters"`
+			Policy      string `json:"policy"`
+		}{
+			Description: "Updated description",
+			Name:        "Updated Policy",
+			Parameters:  "{\"param\":\"value\"}",
+			Policy:      "{\"rule\":\"new_rule\"}",
+		},
+		OwnerUserGroups: &[]int{2},
+		OwnerUsers:      &[]int{2},
+	}
+)
+
+func TestAccResourceAzurePolicyCreate(t *testing.T) {
+	config := testAccAzurePolicyGenerateResourceDeclaration(&initialTestAzurePolicy)
+	hc.PrintHCLConfig(config) // Print the generated HCL configuration
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccAzurePolicyCheckResourceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check:  resource.ComposeTestCheckFunc(testAccAzurePolicyCheckResource(&initialTestAzurePolicy)...),
+			},
+		},
+	})
+}
+
+func TestAccResourceAzurePolicyUpdate(t *testing.T) {
+	initialConfig := testAccAzurePolicyGenerateResourceDeclaration(&initialTestAzurePolicy)
+	updatedConfig := testAccAzurePolicyGenerateResourceDeclaration(&updatedTestAzurePolicy)
+	hc.PrintHCLConfig(initialConfig) // Print the initial HCL configuration
+	hc.PrintHCLConfig(updatedConfig) // Print the updated HCL configuration
```

Comment:
Could this be commented out to produce quieter unit tests and then uncommented when debugging?